services-api/src/main/java/io/scalecube/services/api/ServiceMessage.java

```diff
@@ -132,10 +119,26 @@ public String dataFormat() {
     return header(HEADER_DATA_FORMAT);
   }
 
+  /**
+   * Returns data format of the message data or default one.
+   *
+   * @return data format of the data or default one
+   */
   public String dataFormatOrDefault() {
     return Optional.ofNullable(dataFormat()).orElse(DEFAULT_DATA_FORMAT);
   }
 
+  /**
+   * Returns credentials.
+   *
+   * @param <T> credentials type
+   * @return credentials
+   */
+  public <T> T credentials() {
+    // noinspection unchecked
```

Comment:
it's `noinspection unchecked` because it's an object an not a specific class. try to change it

Reply:
This part of the code doesn't exist anymore since credentials are part of headers.